crates/aptos-crypto/src/multi_ed25519.rs

```diff
@@ -527,16 +527,16 @@ impl Signature for MultiEd25519Signature {
                 )
             ));
         }
-        let mut bitmap_index = 0;
-        // TODO: Eventually switch to deterministic batch verification
-        for sig in &self.signatures {
-            while !bitmap_get_bit(self.bitmap, bitmap_index) {
-                bitmap_index += 1;
-            }
-            sig.verify_arbitrary_msg(message, &public_key.public_keys[bitmap_index])?;
-            bitmap_index += 1;
-        }
-        Ok(())
+
+        self.signatures
+            .iter()
+            .enumerate()
+            .filter(|(bitmap_index, _)| bitmap_get_bit(self.bitmap, *bitmap_index))
+            .map(|(bitmap_index, sig)| {
+                sig.verify_arbitrary_msg(message, &public_key.public_keys[bitmap_index])
+            })
+            .reduce(|accumulator, element| accumulator.and(element))
```

Comment:
How is this more efficient. I don't see how this would short circuit on a failed verification and in fact can increase the costs for bad verifications.

Reply:
Thanks for letting me know! I hadn't preserved the short circuit thing.
Just updated the code using `try_for_each`  to recover the short circuit feature.

My point is, generally, using a functional programming style should help the compiler to do more vectorization and improve efficiency.